### PR TITLE
plugins/comment-nvim: add {pre,post}Hook options

### PR DESCRIPTION
**Pull request by hoppla20**
Adding support for [comment-nvim's hooks](https://github.com/numToStr/Comment.nvim?tab=readme-ov-file#-hooks). These are especially important for integrating comment-nvim with ts-context-commentstring (see linked comment-nvim docs).

### CODE REVIEW

**Comment by hoppla20** [comment]
Hmm okay this is not working. I tried the following:

```nix
{
  plugins.ts-context-commentstring = {
    enable = true;
    extraOptions.enable_autocmd = false;
  };
  plugins.comment-nvim.preHook = "require('ts_context_commentstring.integrations.comment_nvim').create_pre_hook()";
}
```

This should match [ts-context-commentstring documentation](https://github.com/JoosepAlviste/nvim-ts-context-commentstring/wiki/Integrations#commentnvim).

The resulting init.lua looks like this:

```lua
do
  local nixvim_globals = {
   ...
    ["skip_ts_context_commentstring_module"] = true,
    ...
  }

  for k, v in pairs(nixvim_globals) do
    vim.g[k] = v
  end
end


...

require("Comment").setup({
   ["pre_hook"] = "require('ts_context_commentstring.integrations.comment_nvim').create_pre_hook()",
})

...

require("ts_context_commentstring").setup({ ["enable_autocmd"] = false })

...
```

I'm not entirely sure, why this is not working, I just started switching to neovim, so I'm not so familiar how to debug these problems. The following error is printed:

```
[Comment.nvim] Invalid commentstring for nix! Read `:h commentstring` for help.
```

However, when I look at the commentstring it has the following value: `commentstring=# %s`. This should be a valid commentstring, so I am a bit confused. When I use ts-context-commentstring without the hook integration, it works. However, this is suboptimal, because comment-nvim has [some problems without ts-context-commentstring](https://github.com/numToStr/Comment.nvim?tab=readme-ov-file#-treesitter).

Do you maybe have an idea why this is not working?

File: plugins/utils/comment-nvim.nix [hunk 1]
```diff
@@ -29,6 +29,8 @@ in {
         description = "Lines to be ignored while comment/uncomment";
         default = null;
       };
+      preHook = helpers.defaultNullOpts.mkStr "" "Lua function called before (un)comment.";
+      postHook = helpers.defaultNullOpts.mkStr "" "Lua function called after (un)comment.";
```
Comment (traxys) [inline plugins/utils/comment-nvim.nix:33]: I'd suggest using `helpers.defaultNullOpts.mkNullable types.str "null" "...."` as a type instead, to denote the null value.

File: plugins/utils/comment-nvim.nix [hunk 2]
```diff
@@ -94,6 +96,8 @@ in {
   config = let
     setupOptions = {
       inherit (cfg) padding sticky ignore toggler opleader mappings;
+      pre_hook = cfg.preHook;
+      post_hook = cfg.postHook;
```
Comment (traxys) [inline plugins/utils/comment-nvim.nix:100]: you need `pre_hook = helpers.mkRaw cfg.preHook` to make it as lua code (same for the other line)

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/comment-nvim.nix (hunk 2 above)
Comment (hoppla20) [reply plugins/utils/comment-nvim.nix:100]: Thank you, this worked!

**Comment by hoppla20** [comment]
Tested with the above config snippet, should be working now. Maybe in the future one could add a `plugins.ts-context-commentstring.integrations` options, which configures ts-context-commentstring and the corresponding comment plugin (comment-nvim, commentary, ...).

**Review by traxys — CHANGES_REQUESTED** [review]

**Pull request closed without merging**